test/user.test.js

```diff
@@ -1763,6 +1763,367 @@ describe('User', function() {
+          function(next) {
+            User.findById(usersId, function(err, userFound)  {
+              if (err) return next(err);
+              userFound.replaceAttributes({ 'email': currentEmailCredentials.email,
+            'password': currentEmailCredentials.password,
+             'firstName': 'Candy' }, function(err, userInstance) {
+                if (err) return next(err);
+                assert.equal(userInstance.firstName, 'Candy');
+                next();
+              });
+            });
+          },
+          function(next) {
+            AccessToken.find({ where: { userId: usersId }}, function(err, tokens) {
+              if (err) return next(err);
+              expect(tokens.length).to.equal(2);
+              next();
+            });
+          },
+        ], function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `updateOrCreate`', function(done) {
+        async.series([
+          function(next) {
+            User.updateOrCreate({ id: usersId, 'firstName': 'Amir',
+          'email': currentEmailCredentials.email, 'password': 'bar' },
+          function(err, userInstance) {
+            if (err) return next(err);
+            assert.equal(userInstance.firstName, 'Amir');
+            next();
+          });
+          },
+          function(next) {
+            AccessToken.find({ where: { userId: usersId }}, function(err, tokens) {
+              if (err) return next(err);
+              expect(tokens.length).to.equal(2);
+              next();
+            });
+          },
+        ], function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `replaceById`', function(done) {
+        async.series([
+          function(next) {
+            User.replaceById(usersId, { 'firstName': 'Miroslav',
+          'email': currentEmailCredentials.email, 'password': currentEmailCredentials.password },
+            function(err, userInstance) {
+              if (err) return next(err);
+              assert.equal(userInstance.email, currentEmailCredentials.email);
+              next();
+            });
+          },
+          function(next) {
+            AccessToken.find({ where: { userId: usersId }}, function(err, tokens) {
+              if (err) return next(err);
+              expect(tokens.length).to.equal(2);
+              next();
+            });
+          },
+        ], function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+    });
+
+    describe('User not chaning email property', function() {
+      var user1Id, user2Id, user3Id;
+      it('keeps sessions untouched if email property is not changed', function(done) {
+        async.series([
+          function(next) {
+            User.create({ email: 'user1@a.com', password: 'u1pass' }, function(err, user1) {
```

Comment:
Unhandled `err`. Same applies to other places below. Consider using 

```
async.each(
  [
    { email, password },
    //...
  ],
  function(data, next) { User.create(data, next) },
  function(err, users) {
    // ...
  });
});
```

I would personally use promises instead:

```js
var userData = [...];
Promise.all(userData.map(function(data) { return User.create(data) }))
  .then(function(users) {
    // ...
  });
```